lib/yarp.rb

```diff
@@ -165,8 +167,64 @@ def visit_child_nodes(node)
   # This lexes with the Ripper lex. It drops any space events but otherwise
   # returns the same tokens.
   def self.lex_ripper(source)
-    Ripper.lex(source).each_with_object([]) do |token, tokens|
-      tokens << token unless token[1] == :on_sp
+    Ripper.lex(source).each_with_object([]) do |token_arr, tokens|
+      token = LexedToken.new(*token_arr)
```

Comment:
Oh, I wasn't understanding that at all, thanks for explaining.

Happy to do this if you prefer, but to me I think it makes sense to just have a `LexedToken` (or `Token` -- just the name is used elsewhere) type, and do the comparison on that type. It's a little odd to me to have the tokens be sometimes arrays, and sometimes `CommentTokens`. I think _not_ special casing `CommentToken`s also makes the API a little nicer with `LexedToken#is_tilde_heredoc?`. 

I pushed one more stab at this, getting rid of most of the `LexedToken` methods since it's `SimpleDelegator`, but still preserving the named methods. Let me know how that looks to you. If you'd still prefer the arrays, with just a `CommentToken` class, I can definitely change to that too.